internal/kafka/internal/services/kafkatlscertmgmt/secure_storage.go

```diff
@@ -48,18 +51,36 @@ func newSecureStorage(config *config.AWSConfig, automaticCertificateManagementCo
 		return nil, err
 	}
 	return &secureStorage{
-		secretClient: secretClient,
-		secretCache:  secretCache,
-		secretPrefix: config.SecretManager.SecretPrefix,
+		secretClient:     secretClient,
+		secretCache:      secretCache,
+		lock:             sync.Mutex{},
+		storageItemLocks: map[string]*sync.Mutex{},
+		secretPrefix:     config.SecretManager.SecretPrefix,
 	}, nil
 }
 
 func (storage *secureStorage) Lock(ctx context.Context, key string) error {
```

Comment:
From which thread(s) are the Lock and Unlock methods called? is it from the thread that ManageAsync triggers? is it from the reconciler thread or from both?

Reply:
These are called from the ManageAsync which in turn is called from our different reconciler for each Kafka in a given state (`preparing`, `ready` or `provisioning`).